### PR TITLE
correct the openstack.org link and delete some Slash

### PR DESCRIPTION
**Pull request by jianglingxia**
**What this PR does / why we need it**:
correct the metadata-service of openstack.org and delete some slash
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by jianglingxia** [comment]
PTAL @smarterclayton
thanks

File: staging/src/k8s.io/apimachinery/pkg/types/nodename.go [hunk 1]
```diff
@@ -20,23 +20,22 @@ package types
 // Being a type captures intent and helps make sure that the node name
 // is not confused with similar concepts (the hostname, the cloud provider id,
 // the cloud provider name etc)
-//
```
Comment (tallclair) [inline staging/src/k8s.io/apimachinery/pkg/types/nodename.go]: Please revert the changes in this file, they're just noise.

**Comment by dims** [comment]
@jianglingxia Can you please rebase?

File: pkg/cloudprovider/providers/openstack/metadata.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ const (
 	// metadataUrlTemplate allows building an OpenStack Metadata service URL.
 	// It's a hardcoded IPv4 link-local address as documented in "OpenStack Cloud
 	// Administrator Guide", chapter Compute - Networking with nova-network.
-	// https://docs.openstack.org/admin-guide/compute-networking-nova.html#metadata-service
+	// https://docs.openstack.org/nova/pike/admin/networking-nova.html#metadata-service
```
Comment (dims) [inline pkg/cloudprovider/providers/openstack/metadata.go]: can we use `https://docs.openstack.org/nova/latest/admin/networking-nova.html#metadata-service` i.e, `latest` instead of `pike`? thanks!

Comment (jianglingxia) [reply pkg/cloudprovider/providers/openstack/metadata.go]: thanks,I have fix it in pr #59029
